### PR TITLE
[RooFit] RooDataHist Clone

### PR DESCRIPTION
**Pull request by amarini**
Add back default to Clone() for RooDataHist.

### CODE REVIEW

File: roofit/roofitcore/inc/RooDataHist.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ class RooDataHist : public RooAbsData, public RooDirItem {
   RooDataHist& operator=(const RooDataHist&) = delete;
 
   RooDataHist(const RooDataHist& other, const char* newname = 0) ;
-  virtual TObject* Clone(const char* newname) const {
+  virtual TObject* Clone(const char* newname=0) const {
```
Comment (hageboeck) [inline roofit/roofitcore/inc/RooDataHist.h]: Let's use `nullptr` to make it C++11-compliant.

**Comment by hageboeck** [comment]
Could we discuss a bit for which cases this is needed? The problem is as follows:
TObject defines the default as `""`, whereas here it's defined as `0`. That means that if you use a data hist as a DataHist, you get different behaviour than using it as a `TObject*`. That's inconsistent at best.
Would a default of `""` also work?

**Comment by amarini** [comment]
I changed to "" as in TObject. 
A use case is to ensure that the RooDataHist is moved from file to memory and to keep compatibility with the TObject calls.

**Comment by hageboeck** [comment]
Thanks @amarini!

**Pull request merged**